lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -592,13 +595,33 @@ spec = do
                 , "1 +1x + 1y"
                 , "1 + 1x+ 1y"
                 , "1 + 1x +1y"
+                , "xxxx"
+                , "1 + 6667y + 12x"
                 ]
         forM_ invalidFeePolicyTexts $ \policyText ->
             it ("fail fromText @FeePolicy " <> show policyText) $ do
                 let err =
                         "Unable to decode FeePolicy: \
                         \Linear equation not in expected format: a + bx + cy"
                 fromText @FeePolicy policyText === Left (TextDecodingError err)
+        let policyFeesNoFloatingNb =
+                [ "a + bx + cy"
+                , "dasd + asdax + dadsy" ]
+
+        forM_ policyFeesNoFloatingNb $ \policyText ->
+            it ("fail fromText @FeePolicy " <> show policyText) $ do
+                let err = "Expecting floating number"
+                fromText @FeePolicy policyText === Left (TextDecodingError err)
+
+        let correctPolicyTexts =
+                [ "1 + 6667x + 12y"
+                , "1.12 + 1.4324x + 23y"
```

Comment:
I was expecting policy as `a + bx + c` where `a, b, c` are natural numbers and zero, so not sure then if this actually should be deemed as correct?

Reply:
according to the types it is correct : https://github.com/input-output-hk/cardano-wallet/blob/25ab1fab02683590d089061ba232310ee8cc3caf/lib/core/src/Cardano/Wallet/Primitive/Types.hs#L715 Another think to consider is whether this type is proper as we have this https://github.com/input-output-hk/chain-libs/blob/master/chain-impl-mockchain/doc/format.md#type-0-initial-blockchain-configuration `Double` can service properly `Word64` here and endup with proper calculation of `Fee` that is of `Word64` type

Reply:
Also @piotr-iohk The constant fee for certificate is actually a fee "per certificate" (hence the `y`). One can in practice send multiple certificates in one transaction, and have to pay fee for each of them.